Adding repeatable headers to StartRecording

cochi2 opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

cochi2 committed:
Adding repeatable headers for StartRecording

cochi2 committed:
Adding repeatable headers to async recording

cochi2 committed:
Adding export

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.Communication.CallAutomation](https://apiview.dev/Assemblies/Review/8cf6d17c1aae4a5ca44fd55bb28c6286)

minwoolee-msft committed:
- Fixing RepeatabilityHeaders: Allow null to be passed (Do not send repeataiblity headers)
- Auto-create repeatabilityHeaders by default
- Updated CustomeHMACHeaders to allow rawId in HMAC authentication
- Updated all tests and re-recorded based on changes

minwoolee-msft committed:
Updating API doc

minwoolee-msft committed:
Fixing hangup + repeatability headers
Initializing repeatability headers in all options

minwoolee-msft committed:
Fixing Tests + recorded tests

cochi2 committed:
Merge branch 'minwoolee/fixingtests' into fmorales/idempotency

cochi2 committed:
Replacing the null repeatability headers case

cochi2 committed:
Merge branch 'main' into fmorales/idempotency
